metadata-protobuf/proto/App.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto2";
+
+import "proto/Video.proto";
+import "proto/Channel.proto";
+
+message AppActionMetadata {
+  // YouTube video ID
+  optional string video_id  = 2;
+}
+
+message AppAction {
+  // ID of application
+  optional string app_id = 1;
+
+  // Metadata in hex
+  optional string metadata = 2;
+
+  // Raw metadata of wrapped action in hex
+  optional string raw_action = 3;
+
+  // Signature over app commitment
+  optional string signature = 4;
```

Comment:
I just realized that having those parameters provided as hex is quite suboptimal. You need 2 hex characters to represent one byte, so there are at least 2 bytes used to represent a single byte of data at the moment.

There is a `bytes` type in protobuf, perhaps you can give it a try? `protobufjs` represents it as `UInt8Array` in JavaScript